presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroColumnDecoder.java

```diff
@@ -244,13 +251,13 @@ private static void serializePrimitive(BlockBuilder blockBuilder, Object value,
             return;
         }
 
-        if (type instanceof BigintType) {
-            type.writeLong(blockBuilder, (Long) value);
+        if (type instanceof BigintType || type instanceof IntegerType || type instanceof SmallintType || type instanceof TinyintType) {
+            type.writeLong(blockBuilder, ((Number) value).longValue());
```

Comment:
`Number#longValue` is a code smell, as it can truncate/round the contained value

retain the cast to `Long`, since the other integer types are represented in Presto by a long too

Reply:
Reverted: the value will either be a java Long or Integer as it is coming from avro LONG and INT. I will add unit tests for these as well.

Reply:
Had to modify this and add back the Number#longValue - verifying that the value is a java Integer or Long since that is what comes from avro.

Reply:
> the value is a java Integer or Long since that is what comes from avro

i missed that. Okay, so use this information. it's really recommended to stay clear from `Number#longValue`